### PR TITLE
Handle invalid datetimes in dataframe_from_result_table

### PR DESCRIPTION
**Pull request by liamkirton**
This is a fix for #355, where dataframe_from_result_table fails to handle out-of-range datetime conversion.

### CODE REVIEW

**Comment by AsafMah** [comment]
Looks good, can you add a test?

File: azure-kusto-data/tests/test_helpers.py [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+    import pandas
+    import numpy
+
+    PANDAS = True
+except:
+    pass
+
+
+# Sample response containing conversion edge cases.
+RESPONSE_TEXT = """
```
Comment (AsafMah) [inline azure-kusto-data/tests/test_helpers.py:22]: Move this to a json file like we do in other tests

File: azure-kusto-data/tests/test_helpers.py [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+},
+{
+    "FrameType": "DataSetCompletion",
+    "HasErrors": false,
+    "Cancelled": false
+}]
+"""
+
+
+class TestDataFrameFromResultsTable(unittest.TestCase):
```
Comment (AsafMah) [inline azure-kusto-data/tests/test_helpers.py:213]: We use pytest, so please remove the inheritance from unittest.TestCase, as it may interfere

**Review by AsafMah — COMMENTED** [review]
Looks good, just some small changes

File: azure-kusto-data/tests/test_helpers.py (hunk 1 above)
Comment (liamkirton) [reply azure-kusto-data/tests/test_helpers.py]: Done!

File: azure-kusto-data/tests/test_helpers.py (hunk 2 above)
Comment (liamkirton) [reply azure-kusto-data/tests/test_helpers.py]: Done!

**Comment by AsafMah** [comment]
Can you see why the test fails?

**Comment by liamkirton** [comment]
The test failure on 3.6 was due to differences in Pandas behaviour: between the version on 3.6 and 3.7, the minimum representable datetime changes slightly...

pd._libs.tslibs.timestamps.Timestamp.min == Timestamp('1677-09-21 00:12:43.145225')
vs.
pd._libs.tslibs.timestamps.Timestamp.min == Timestamp('1677-09-21 00:12:43.145224193')

I've relaxed the test criteria to just check that parsing a close-to-min/max timestamp succeeds.

**Review by AsafMah — APPROVED** [review]
